content/help/ops/traffic-management/deploy-guidelines/index.md

```diff
@@ -0,0 +1,205 @@
+---
+title: Deployment and Configuration Guidelines
+description: Provides specific deployment or configuration guidelines or issue avoidance
+weight: 5
+---
+
+This section provides specific deployment or configuration guidelines to avoid networking or traffic management issues.
+
+## 503 errors while reconfiguring service routes
+
+When setting route rules to direct traffic to specific versions (subsets) of a service, care must be taken to ensure
+that the subsets are available before they are used in the routes. Otherwise, calls to the service may return
+503 errors during a reconfiguration period.
+
+Creating both the `VirtualServices` and `DestinationRules` that define the corresponding subsets using a single `kubectl`
+call (e.g., `kubectl apply -f myVirtualServiceAndDestinationRule.yaml` is not sufficient because the
+resources propagate (from the configuration server, i.e., Kubernetes API server) to the Pilot instances in an eventually consistent manner. If the
+`VirtualService` using the subsets arrives before the `DestinationRule` where the subsets are defined, the Envoy configuration generated by Pilot would refer to non-existent upstream pools, resulting in HTTP 503 errors until all configuration objects are available to Pilot.
+
+To make sure services will have zero down-time when configuring routes with subsets, follow a "make-before-break" process as described below:
+
+* When adding new subsets:
+
+    1. Update `DestinationRules` to add a new subset first, before updating any `VirtualServices` that use it. Apply the rule using `kubectl` or any platform-specific tooling.
+
+    1. Wait a few seconds for the `DestinationRule` configuration to propagate to the Envoys
+
+    1. Update the `VirtualService` to refer to the newly added subsets.
+
+* When removing subsets:
+
+    1. Update `VirtualServices` to remove any references to a subset, before removing the subset from a `DestinationRule`.
+
+    1. Wait a few seconds for the `VirtualService` configuration to propagate to the Envoys
```

Comment:
Envoys -> Envoy side cars.

Reply:
Another option is to use language such as "propagate within the Envoy data plane."